src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -530,6 +530,60 @@ StorageReplicatedMergeTree::StorageReplicatedMergeTree(
     initialization_done = true;
 }
 
+PartMetadataFormatVersion StorageReplicatedMergeTree::partMetadataFormatVersion() const
+{
+    auto min = PartMetadataFormatVersion{getSettings()->desired_part_metadata_format_version};
+    if (min == PART_METADATA_FORMAT_VERSION_OLD)
+        // this is the lowest possible value. in this case, other replicas' opinion will not matter
+        return min;
+
+    try {
+        // TODO Consider using watches, and compute the correct format version in the background
+        // but it seems maybe overkill?
```

Comment:
I'm open to suggestions on whether this lookup of sibling selected versions should happen in a background thread, or if it's ok to do it here. This function will be called anytime we want to create a new part, so I'm thinking maybe the extra back-and-forth with keeper _might_ be problematic?